src/liblsquic/lsquic_rechist.c

```diff
@@ -219,6 +219,8 @@ rechist_alloc_elem (struct lsquic_rechist *rechist)
     idx = find_free_slot(*mask);
     *mask |= 1ull << idx;
     ++rechist->rh_n_used;
+    rechist->rh_elems[idx + i * BITS_PER_MASK].re_next  = UINT_MAX;
```

Comment:
Is this line necessary? seems it does not change the test result.

Reply:
It doesn't change the result, but it seems good practice to me that objects should be properly initialised when created. A fresh elem should have the next pointer set to NULL, shouldn't it? Belt and braces. :)

Reply:
In general, yes, but, in this case, that value is initialized by the caller immediately after calling `rechist_alloc_elem()`, so it is a duplicate effort.